testsuite/tests/libratest/smoke_test.rs

```diff
@@ -1312,3 +1327,194 @@ fn test_key_manager_consensus_rotation() {
         );
     }
 }
+
+struct AccountContext {
+    private_key: Ed25519PrivateKey,
+    address: AccountAddress,
+}
+
+/// Submits a transaction, and returns appropriate account and sequence number for verification
+fn submit_new_transaction(
+    libra: &JsonRpcLibraInterface,
+    time_service: &RealTimeService,
+    account_context: &AccountContext,
+    script: Script,
+) -> anyhow::Result<(AccountAddress, u64)> {
+    let sequence_number = libra.retrieve_sequence_number(account_context.address)?;
+    let expiration_time = time_service.now() + constants::TXN_EXPIRATION_SECS;
+    let raw_txn = RawTransaction::new_script(
+        account_context.address,
+        sequence_number,
+        script,
+        constants::MAX_GAS_AMOUNT,
+        constants::GAS_UNIT_PRICE,
+        constants::GAS_CURRENCY_CODE.to_owned(),
+        Duration::from_secs(expiration_time),
+    );
+
+    // Sign and submit the transaction
+    let signature = account_context.private_key.sign_message(&raw_txn.hash());
+    let signed_txn =
+        SignedTransaction::new(raw_txn, account_context.private_key.public_key(), signature);
+    libra.submit_transaction(Transaction::UserTransaction(signed_txn))?;
+    Ok((account_context.address, sequence_number))
+}
+
+fn submit_new_validator_config(
+    libra: &JsonRpcLibraInterface,
+    time_service: &RealTimeService,
+    account_context: &AccountContext,
+    validator_config: ValidatorConfig,
+) -> anyhow::Result<(AccountAddress, u64)> {
+    // Convert validator_config into an update transaction
+    let consensus_key = validator_config.consensus_public_key;
+    let validator_network_address = validator_config.validator_network_address;
+    let validator_network_key = validator_config.validator_network_identity_public_key;
+    let full_node_network_key = validator_config.full_node_network_identity_public_key;
+    let full_node_network_address = validator_config.full_node_network_address;
+    let script = transaction_builder::encode_set_validator_config_script(
+        account_context.address,
+        consensus_key.to_bytes().to_vec(),
+        validator_network_key.to_bytes(),
+        validator_network_address.into(),
+        full_node_network_key.to_bytes(),
+        full_node_network_address.into(),
+    );
+    submit_new_transaction(libra, time_service, account_context, script)
+}
+
+fn submit_new_reconfig(
+    libra: &JsonRpcLibraInterface,
+    time_service: &RealTimeService,
+    account_context: &AccountContext,
+) -> anyhow::Result<(AccountAddress, u64)> {
+    let script = Script::new(
+        libra_transaction_scripts::RECONFIGURE_TXN.clone(),
+        vec![],
+        vec![],
+    );
+    submit_new_transaction(libra, time_service, account_context, script)
+}
+
+/// Helper function to build libra interfaces for smoke test
+fn get_libra_interface(node_config: &NodeConfig) -> JsonRpcLibraInterface {
```

Comment:
this probably should be part of the TestEnvironment or LibraSwarm, given an index return the Interface

Reply:
Thanks for the suggestions!  I think these are great followups for the helper functions of where they can be leveraged to help further development, once I get done with the next thing, I'll see if I can circle back and poke a bit at them.  But, it's not really high priority right now.

Reply:
it's just nit, no worries